diff --git a/osfmk/i386/machine_routines.h b/osfmk/i386/machine_routines.h
index 5e1a446..b15eb58 100644
--- a/osfmk/i386/machine_routines.h
+++ b/osfmk/i386/machine_routines.h
@@ -80,7 +80,7 @@ void ml_install_interrupt_handler(
 
 void ml_entropy_collect(void);
 
-uint64_t ml_get_timebase();
+uint64_t ml_get_timebase(void);
 void ml_init_lock_timeout(void); 
 void ml_init_delay_spin_threshold(int);
 
